packages/venia-concept/src/actions/checkout/asyncActions.js

```diff
@@ -27,6 +30,7 @@ export const cancelCheckout = () =>
 export const resetCheckout = () =>
     async function thunk(dispatch) {
         await dispatch(closeDrawer());
+        await dispatch(createGuestCart());
```

Comment:
Consider guarding this with a check to see if `guestCartId` exists in state.

Reply:
Good call, I actually purposefully didn't do that because `createGuestCart` does it itself:

```js
export const createGuestCart = () =>
    async function thunk(dispatch, getState) {
        const { cart } = getState();

        // if a guest cart already exists, exit
        if (cart.guestCartId) {
            return;
        }

// more code
```